### PR TITLE
Fix all warnings.

### PR DESCRIPTION
**Pull request by collijk**
This PR fixes all the sphinx build warnings and sets the documentation to fail if new docs cause warnings.  This should help make sure all links resolve properly and catch a lot of formatting errors automatically.

### CODE REVIEW

**Review by yongqx — APPROVED** [review]

File: docs/source/gbd2019_models/interventions/index.rst [hunk 1]
```diff
@@ -1,16 +0,0 @@
-.. _2019_intervention_models:
```
Comment (beatrixkh) [inline docs/source/gbd2019_models/interventions/index.rst:1]: Why was this one deleted?

File: docs/source/gbd2017_models/interventions/index.rst [hunk 2]
```diff
@@ -1,16 +0,0 @@
-.. _2017_intervention_models:
```
Comment (beatrixkh) [inline docs/source/gbd2017_models/interventions/index.rst:1]: and this one?

**Review by beatrixkh — APPROVED** [review]
Holy crap james thank you this is a ton of needling work. Looks good! much appreciated!!

File: docs/source/gbd2019_models/interventions/index.rst (hunk 1 above)
Comment (collijk) [reply docs/source/gbd2019_models/interventions/index.rst:1]: No documents in these directories have ever been made

Comment (beatrixkh) [reply docs/source/gbd2019_models/interventions/index.rst:1]: 🤷 okee this makes sense

**Review by alibow — APPROVED** [review]
Thanks so much James!!

File: docs/source/gbd2017_models/concept_models/vivarium_conceptmodel_template/concept_model.rst [hunk 3]
```diff
@@ -31,11 +45,13 @@
...
 
-======================
-Concept model template
-======================
+
+.. _{YOUR_MODEL_UNDERSCORE}:
+
+==================
+{YOUR_MODEL_TITLE}
+==================
```
Comment (NathanielBlairStahn) [inline docs/source/gbd2017_models/concept_models/vivarium_conceptmodel_template/concept_model.rst:54]: Will this make the "Concept model template" page show up as "{YOUR_MODEL_TITLE}" in the index?

Also, does this page even show up in the index?

File: docs/source/gbd2017_models/index.rst [hunk 4]
```diff
@@ -8,5 +8,5 @@ GBD 2017 Models
    :maxdepth: 2
    :glob:
 
-   *
    */index
+   concept_models/vivarium_conceptmodel_template/concept_model
```
Comment (NathanielBlairStahn) [inline docs/source/gbd2017_models/index.rst:12]: Oh, I guess that answers my question above about whether the page shows up in the index.

**Review by NathanielBlairStahn — APPROVED** [review]

**Pull request merged**